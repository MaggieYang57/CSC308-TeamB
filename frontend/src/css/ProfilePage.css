.header {
  margin-bottom: 10vw;
}

h1 {
  margin-left: 10vw;
  letter-spacing: 2px;
  font-weight: 700;
  font-size: 40px;
  color: #265b5f;
}

h2 {
  font-size: 25px;
  font-weight: 800;
  color: #265b5f;
}

h3 {
  justify-content: center;
  text-align: center;
  font-family: "Mulish", sans-serif;
  font-size: 25px;
  color: #265b5f;
}

.saved-trails {
  margin-top: 2vw;
  margin-bottom: 10vw;
}

.saved-title hr {
  display: inline-block;
  width: 30%;
  margin: 5px 10px;
  border-top: 1px solid #265b5f;
}
