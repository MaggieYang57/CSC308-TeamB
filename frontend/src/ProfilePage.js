import React from "react";
import ReviewTable from "./components/ReviewTable";
import "bootstrap/dist/css/bootstrap.min.css";
import "./css/ProfilePage.css";

require('dotenv').config()
const backendHostURL = process.env.REACT_APP_BACKEND_HOST_URL

class ProfilePage extends React.Component {
  constructor(props) {
    super(props);
    this.state = {};
  }

  componentDidMount() {
    fetch(`${backendHostURL}/login/${localStorage.getItem("_id")}`)
      .then((res) => res.json())
      .then((data) => {
        console.log("data", data);
        this.setState({ ...data[0] });
      });
  }

  render() {
    return (
      <div>
        <h1 className="header">
          Hello, {this.state.first_name} {this.state.last_name}
        </h1>
        <div className="saved-trails">
          <h2 className="saved-title">
            <hr />
            Saved Trails
            <hr />
          </h2>
          <h3>{(this.state.saved_trails || []).map(item => (
            <li key={item} style={{ marginBottom: "10px" }}>{item}</li>
          ))}</h3>
        </div>
        <div className="reviews">
          <h2 className="reviews-title">
            <hr />
            Reviews
            <hr />
          </h2>
          <ReviewTable reviewList={this.state.user_email} route={"user"} />
        </div>
      </div>
    );
  }
}

export default ProfilePage;
