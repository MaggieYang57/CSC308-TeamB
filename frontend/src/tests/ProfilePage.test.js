// eslint-disable-next-line no-unused-vars
import AdminUserReviewTable from "../components/AdminUserReviewTable";
import { cleanup } from '@testing-library/react';
import React from 'react';
import { MemoryRouter } from 'react-router';
import renderer from 'react-test-renderer';
import ProfilePage from '../ProfilePage';
import createFetchSpy from './CreateFetchSpy';

jest.mock('../components/AdminUserReviewTable', () => 'hi');

describe('Profile Page', () => {

  beforeEach(() => {
    createFetchSpy({}, [
      {
        '_id': 1, 
        'rating': [5, 3, 4], 
        'title':'Hike', 
        'location': 'SLO', 
        'description': 'great hike', 
        'imagesrc': 'test',
        difficulty: [5],
        saved_trails: ['lololol'],
      }
    ]);

  });

  afterEach(() => {
    cleanup();
  });

  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter> <ProfilePage /> </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
