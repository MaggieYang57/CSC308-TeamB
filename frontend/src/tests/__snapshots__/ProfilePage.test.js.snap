// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Page renders correctly 1`] = `
Array [
  " ",
  <div>
    <h1
      className="header"
    >
      Hello, 
       
    </h1>
    <div
      className="saved-trails"
    >
      <h2
        className="saved-title"
      >
        <hr />
        Saved Trails
        <hr />
      </h2>
      <div
        className="container"
      >
        <div
          className="col"
        />
      </div>
    </div>
    <div
      className="reviews"
    >
      <h2
        className="reviews-title"
      >
        <hr />
        Reviews
        <hr />
      </h2>
      <hi
        route="user"
      />
    </div>
  </div>,
  " ",
]
`;
